packages/fxa-shared/package.json

```diff
@@ -106,7 +107,7 @@
     "hot-shots": "^9.0.0",
     "ioredis": "^4.28.2",
     "ip": "^1.1.5",
-    "joi": "^14.3.1",
+    "joi": "17.4.0",
```

Comment:
Interesting. I wonder if **joi** is technically needed as a dependency if we're using **celebrate** here:

> ### Joi
> celebrate exports the version of joi it is using internally. For maximum compatibility, you should use this version when creating schemas used with celebrate.
> &mdash; via https://github.com/arb/celebrate#joi

Where **celebrate@15.0.1** (current "latest") seems to be using **joi@17.x.x**, per https://github.com/arb/celebrate/blob/b4e4c97f005711286830f24030203d5acef0ab46/package.json#L33.

Reply:
🤔 How would files that currently import joi do so if it's being exported by another dependency?

Reply:
Ah yes, this does look mildly trickier (and made more complex by being in /fxa-shared/).

```sh
git grep "celebrate" packages/fxa-shared | cat

packages/fxa-shared/express/routing.js:    celebrate,
packages/fxa-shared/express/routing.js:  } = require('celebrate');
packages/fxa-shared/express/routing.js:          celebrate(routeDefinition.validate, {

packages/fxa-shared/package.json:    "celebrate": "^10.0.1",

packages/fxa-shared/test/express/routing.js:  let celebrateHandler;
packages/fxa-shared/test/express/routing.js:  let celebrateMock;
packages/fxa-shared/test/express/routing.js:    celebrateHandler = () => {};
packages/fxa-shared/test/express/routing.js:    celebrateMock = {
packages/fxa-shared/test/express/routing.js:      celebrate: () => celebrateHandler,
packages/fxa-shared/test/express/routing.js:      celebrate: celebrateMock,
packages/fxa-shared/test/express/routing.js:          celebrateHandler,
packages/fxa-shared/test/express/routing.js:          celebrateHandler,
packages/fxa-shared/test/express/routing.js:    it('logs and delegates validation errors to celebrate', () => {
```

So, my very ignorant grepping shows that **celebrate** is only used in that express/routing.js file, which doesn't also include Joi.

**joi** (or **typesafe-joi**) is used elsewhere in /fxa-shared/, but...  yeah.

```sh
git grep "joi" packages/fxa-shared/ | grep -E "(require\(|import )"

packages/fxa-shared/dto/auth/payments/coupon.ts:import joi from 'typesafe-joi';
packages/fxa-shared/dto/auth/payments/invoice.ts:import joi from 'typesafe-joi';
packages/fxa-shared/express/remote-address.js:const joi = require('joi');
packages/fxa-shared/metrics/navigation-timing-validation.js:const joi = require('joi');
packages/fxa-shared/sentry/report-validation-error.ts:import { ValidationError } from 'joi';
packages/fxa-shared/subscriptions/configuration/base.ts:import joi from 'joi';
packages/fxa-shared/subscriptions/configuration/plan.ts:import joi from 'joi';
packages/fxa-shared/subscriptions/configuration/product.ts:import joi from 'joi';
```

So probably just ignore it, since I'm assuming using something like `const { Joi } = require('celebrate');` might be more trouble than it's worth.
Although now I'm fairly intrigued by **typesafe-joi**.